cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -52,23 +46,19 @@
 public class SecureStoreHandler extends AbstractAppFabricHttpHandler {
   private static final Gson GSON = new Gson();
 
-  private final SecureStore secureStore;
-  private final SecureStoreManager secureStoreManager;
+  private final SecureStoreService secureStoreService;
 
   @Inject
-  public SecureStoreHandler(SecureStore secureStore, SecureStoreManager secureStoreManager) {
-    this.secureStore = secureStore;
-    this.secureStoreManager = secureStoreManager;
+  public SecureStoreHandler(SecureStoreService secureStoreService) {
+    this.secureStoreService = secureStoreService;
   }
 
   @Path("/keys/{key-name}")
```

Comment:
I just noticed: If you have a ``securekeys`` component in the path, then I don't think we need an additional ``keys`` component. Applies to all APIs.